cmd/jsonnet.cpp

```diff
@@ -673,38 +685,76 @@ int main(int argc, const char **argv)
 
             case FMT: {
                 std::string output_file = config.outputFile;
-                if (config.fmtInPlace) {
-                    if (config.inputFile == "-") {
-                        std::cerr << "ERROR: Cannot use --in-place with stdin" << std::endl;
-                        jsonnet_destroy(vm);
-                        return EXIT_FAILURE;
+
+                if (config.fmtInPlace || config.fmtTest) {
+                    assert(config.inputFiles.size() >= 1);
+                    for (std::string inputFile: config.inputFiles) {
+                        if (config.fmtInPlace) {
+                            output_file = inputFile;
+
+                            if (inputFile == "-") {
```

Comment:
> Seems useful to pipe files through the formatter rather than modify them in place.
Of course. It's only preventing specifying the input as "-" when changing the files in place (see the `if` above).
It's still possible to do `jsonnet fmt - < somefile > output_file`, it only prevents `jsonnet fmt --in-place -` which doesn't make any sense.